src/cortexpy/tesserae.py

```diff
@@ -136,8 +159,19 @@ def __align_all(self, panel):
         lsize_l = np.log(size_l)
 
         max_r, pos_max, state_max, who_max = self.__initialization(query, panel, lsize_l)
-        max_r, pos_max, state_max, who_max = self.__recurrence(query, panel, lsize_l, l1, max_r, pos_max, state_max, who_max)
-        cp = self.__termination(l1, pos_max, state_max, who_max)
+        max_r, pos_max, state_max, who_max = self.__recurrence(query, panel, lsize_l, l1, max_r, pos_max, state_max, who_max, store_states=self.mem_limit)
+        cp, pos_max, who_max, state_max = self.__termination(np.remainder(self.qlen, self.traceback_limit) if self.mem_limit else l1, pos_max, state_max, who_max, 2*self.maxl)
+        l2 = self.traceback_limit
+        if self.mem_limit and self.saved_states:
+            self.saved_states.pop()
+        while self.mem_limit and self.saved_states:
+            max_r, pos_max_n, state_max_n, who_max_n, pos_target_n = self.saved_states.pop()
+            idx = len(self.saved_states)
+            self.vt_m[1] = np.copy(self.saved_vt_m[idx])
+            self.vt_i[1] = np.copy(self.saved_vt_i[idx])
+            self.vt_d[1] = np.copy(self.saved_vt_d[idx])
```

Comment:
If I am not mistaken, then encapsulating these matrices in classes would also allow us to separate the storage and retrieval of matrix data from the alignment code.